citrination_client/data/client.py

```diff
@@ -141,6 +145,45 @@ def upload(self, dataset_id, source_path, dest_path=None):
         else:
             raise ValueError("No file at specified path {}".format(source_path))
 
+    def upload_with_ingester(self, dataset_id, source_path, ingester, ingester_arguments=[], dest_path=None):
+        """
+        Upload a file, specifying source and dest paths a file (acts as the scp command).asdfasdf
+
+        :param dataset_id: The ID of the dataset to search for files.
+        :type dataset_id: Union[int, str]
+        :param source_path: The path to the file on the source host asdf
+        :type source_path: str
+        :param ingester: The ingester being used
+        :type ingester: class:`citrination_client.data.ingest.Ingester`
+        :param ingester_arguments: ingester arguments (optional), arguments should
+                                   contain keys "name" and "value"
+        :type ingester_arguments: list of dict
+        :param dest_path: The path to the file where the contents of the upload will be written (on the dest host)
+        :type dest_path: str
+        :return: The result of the upload process
+        :rtype: :class:`UploadResult`
+        """
+        if not os.path.isfile(source_path):
+            raise ValueError("source_path parameter must point to a file".format(source_path))
+
+        self._ingest.validate_ingester(ingester, ingester_arguments)
+        file_upload_result = self.upload(dataset_id, source_path, dest_path)
+
+        if len(file_upload_result.successes) == 1:
+            file_path = file_upload_result.successes[0]["dest_path"]
+            self._ingest.submit(dataset_id, file_path, ingester, ingester_arguments)
+
+        return file_upload_result
+
+    def list_ingesters(self):
```

Comment:
Why do you not just access the child client directly?

Reply:
I was originally going to have direct interactions with the `IngestClient`, but I was unable to encompass all of the upload logic within `IngestClient#submit` due to its need to leverage `DataClient#upload`. I talked with @genfx999 I think, and we concluded that we might as well have ingester related actions go through the `DataClient` rather than have users use the `IngestClient` solely for its `list_ingesters` method.